src/pcp.rs

```diff
@@ -179,11 +181,15 @@ pub enum PcpError {
 
 /// A value of a certain type. Implementations of this trait specify an arithmetic circuit that
 /// determines whether a given value is valid.
-pub trait Value<F, G>: Sized + PartialEq + Eq + Debug
+pub trait Value<F, G>:
+    Sized + PartialEq + Eq + Debug + TryFrom<(<Self as Value<F, G>>::Param, Vec<F>), Error = TypeError>
```

Comment:
`TryFrom<T>` where `T` is a tuple is interesting. How does the [implied `TryInto`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html#generic-implementations) implementation work? You get back the `Vec<F>` and the `Param`?

Reply:
Hmm, I hadn't considered that. Yes, ideally you'd get the `(Vec<F>, Param)`, but I haven't tried this.